python/gaas_client/types.py

```diff
@@ -12,9 +12,72 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+import numpy
+
 from gaas_client.gaas_thrift import spec
 
 Value = spec.Value
 DataframeRowIndex = spec.DataframeRowIndex
 BatchedEgoGraphsResult = spec.BatchedEgoGraphsResult
 Node2vecResult = spec.Node2vecResult
+UniformNeighborSampleResult = spec.UniformNeighborSampleResult
+
+
+class UnionWrapper:
+    """
+    Provides easy conversions between py objs and Thrift "unions".
+    """
+    def get_py_obj(self):
+        not_members = set(["default_spec", "thrift_spec", "read", "write"])
+        attrs = [a for a in dir(self.union)
+                    if not(a.startswith("_")) and a not in not_members]
+        for a in attrs:
+            val = getattr(self.union, a)
+            if val is not None:
+                return val
+
+        return None
+
+
+class ValueWrapper(UnionWrapper):
+    def __init__(self, val, val_name="value"):
+        if isinstance(val, Value):
+            self.union = val
+        elif isinstance(val, int):
+            if val < 4294967296:
+                self.union = Value(int32_value=val)
+            else:
+                self.union = Value(int64_value=val)
+        elif isinstance(val, numpy.int32):
+            self.union = Value(int32_value=int(val))
+        elif isinstance(val, numpy.int64):
+            self.union = Value(int64_value=int(val))
+        elif isinstance(val, str):
+            self.union = Value(string_value=val)
+        elif isinstance(val, bool):
+            self.union = Value(bool_value=val)
+        else:
+            raise TypeError(f"{val_name} must be one of the "
+                            "following types: [int, str, bool], got "
+                            f"{type(val)}")
+
+
+class DataframeRowIndexWrapper(UnionWrapper):
+    def __init__(self, val, val_name="dataframe_row_index"):
+        if isinstance(val, DataframeRowIndex):
+            self.union = val
+        elif isinstance(val, int):
+            if val >= 4294967296:
+                self.union = DataframeRowIndex(int64_index=val)
+            else:
+                self.union = DataframeRowIndex(int632_index=val)
```

Comment:
```suggestion
                self.union = DataframeRowIndex(int32_index=val)
```

Reply:
Thanks! This should have been caught in testing so I added some tests, confirmed it caught this typo, then fixed it.